applications/CompressiblePotentialFlowApplication/custom_processes/define_embedded_wake_process.cpp

```diff
@@ -0,0 +1,143 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:        BSD License
+//                  Kratos default license: kratos/license.txt
+//
+//  Main authors:   Marc Nunez
+//
+
+
+#include "define_embedded_wake_process.h"
+#include "processes/calculate_discontinuous_distance_to_skin_process.h"
+#include "compressible_potential_flow_application_variables.h"
+#include "custom_utilities/potential_flow_utilities.h"
+
+
+namespace Kratos
+{
+// Constructor for DefineEmbeddedWakeProcess Process
+DefineEmbeddedWakeProcess::DefineEmbeddedWakeProcess(ModelPart& rModelPart,
+                    ModelPart& rWakeModelPart
+                ):
+    Process(),
+    mrModelPart(rModelPart),
+    mrWakeModelPart(rWakeModelPart)
+{}
+
+void DefineEmbeddedWakeProcess::Execute()
+{
+    KRATOS_TRY;
+
+    KRATOS_ERROR_IF(mrModelPart.GetProcessInfo()[DOMAIN_SIZE]>2) << "DOMAIN_SIZE is greater than 2. DefineEmbeddedWakeProcess is only implemented for 2D cases!" << std::endl;
+
+    ComputeDistanceToWake();
+    MarkWakeElements();
+    ComputeTrailingEdgeNode();
+    MarkKuttaWakeElements();
+
+    KRATOS_CATCH("");
+}
+
+
+void DefineEmbeddedWakeProcess::ComputeDistanceToWake(){
+
+    CalculateDiscontinuousDistanceToSkinProcess<2> distance_calculator(mrModelPart, mrWakeModelPart);
```

Comment:
Why did you decide to use this process versus another calculate distance process?

Reply:
In short, discontinous processes are used for volume-less bodies. I have sent you an email to showcase this.